### PR TITLE
Add TypeScript docs

### PR DESCRIPTION
**Pull request by nbeloglazov**
* Add section about running TypeScript unit tests.
* Add best practices for TypeScript.,
* Add more info on TypeScript code setup (organization, bundling).

### CODE REVIEW

File: docs/contributor-guide/developer-guide/getting-started.md [hunk 1]
```diff
@@ -141,28 +141,32 @@ If you'd like to run a specific test or set of tests, and/or save sbt startup ti
...
 
 To run the browser tests (includes all the [Playwright](https://playwright.dev) tests in [`civiform/browser-test/src/`](https://github.com/civiform/civiform/tree/main/browser-test/src), there are three steps:
 
-1.  Build the Docker image for running the playwright tests. This only needs to be done once:
-
-    ```
-    bin/build-browser-tests
-    ```
-2.  Bring up the local test environment using the AWS emulator. This step can be done in a separate terminal window while the above step is still building. Leave this running while you are working for faster browser test runs:
+1.  Bring up the local test environment using the AWS emulator. This step can be done in a separate terminal window while the above step is still building. Leave this running while you are working for faster browser test runs:
```
Comment (shanemc-goog) [inline docs/contributor-guide/developer-guide/getting-started.md:163]: The second sentence is incorrect as it references the step you removed.

I assume that step is in fact not actually needed anymore?

File: docs/contributor-guide/developer-guide/new-full-time-swe-onboarding-materials.md [hunk 2]
```diff
@@ -8,13 +8,22 @@
...
 ### Recommended Tutorials
 
-* [git](https://docs.github.com/en/get-started/using-git/about-git)
-* [docker](https://docs.docker.com/get-started/)
-* [github](https://lab.github.com/githubtraining/introduction-to-github)
-* [play framework (java version)](https://www.playframework.com/documentation/2.8.x/JavaHome)
-* [java guice framework](https://github.com/google/guice/wiki/GettingStarted)
-* [terraform framework tutorial](https://learn.hashicorp.com/tutorials/terraform/infrastructure-as-code)
-* [ebean](https://ebean.io)
+* general
```
Comment (shanemc-goog) [inline docs/contributor-guide/developer-guide/new-full-time-swe-onboarding-materials.md:11]: Init capitalize the headers

File: docs/contributor-guide/developer-guide/new-full-time-swe-onboarding-materials.md [hunk 3]
```diff
@@ -25,13 +34,14 @@
...
 * Slack
 * Added to the technical google group
-* LoginRadius
 * DockerHub (if at Google, need to pay for the professional one)
 * Github Access
   * [Civiform Org](https://github.com/civiform)
   * civiform/civiform
-  * civiform/docs 
+  * civiform/docs
+* Azure (optional, if working on deployment)
```
Comment (shanemc-goog) [inline docs/contributor-guide/developer-guide/new-full-time-swe-onboarding-materials.md:44]: Make the Deployment/optional part a header and then nest these 4 under it and remove the repetition.

File: docs/contributor-guide/developer-guide/technology-overview.md [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 ### Overview
 
-CiviForm is built on the [Play Framework](https://www.playframework.com) in Java, and backed by a [PostgreSQL](https://www.postgresql.org) database. We use Guice to do dependency injection and follow the [Strategy Pattern](https://www.tutorialspoint.com/design\_pattern/strategy\_pattern.htm). The front end uses [J2Html](https://j2html.com) Java library to render HTML (server-side). For applicant authentication we support a few different OIDC/SAML identity providers. For infrastructure we use terraform.
+CiviForm is built on the [Play Framework](https://www.playframework.com) in Java, and backed by a [PostgreSQL](https://www.postgresql.org) database. We use Guice to do dependency injection and follow the [Strategy Pattern](https://www.tutorialspoint.com/design\_pattern/strategy\_pattern.htm). The front end uses [J2Html](https://j2html.com) Java library to render HTML (server-side). To client-side interactions we use TypeScript without using any frameworks (see [Frontend development](frontend-development.md) for more info). For applicant authentication we support a few different OIDC/SAML identity providers. For infrastructure we use terraform.
```
Comment (shanemc-goog) [inline docs/contributor-guide/developer-guide/technology-overview.md:7]:
```suggestion
CiviForm is built on the [Play Framework](https://www.playframework.com) in Java, and backed by a [PostgreSQL](https://www.postgresql.org) database. We use Guice to do dependency injection and follow the [Strategy Pattern](https://www.tutorialspoint.com/design\_pattern/strategy\_pattern.htm). The front end uses [J2Html](https://j2html.com) Java library to render HTML (server-side). For client-side interactions we use TypeScript without using any frameworks (see [Frontend development](frontend-development.md) for more info). For applicant authentication we support a few different OIDC/SAML identity providers. For infrastructure we use terraform.
```

File: docs/contributor-guide/developer-guide/technology-overview.md [hunk 5]
```diff
@@ -16,32 +16,6 @@ All view classes should extend [`BaseHtmlView`](https://github.com/civiform/civi
 
 The `View` classes are generally organized by which role(s) they are viewable by (e.g., [app/view/admin/](https://github.com/civiform/civiform/tree/main/server/app/views/admin) for pages viewable by Admins, [app/views/applicant/](https://github.com/civiform/civiform/tree/main/server/app/views/applicant) for pages viewable by Applicants). Each of these roles also has its own [`Layout` class](https://github.com/civiform/civiform/blob/main/server/app/views/admin/AdminLayout.java) that extends `BaseHtmlLayout` for rendering page content in the context of that role.
 
-## How to add WebJar dependencies
```
Comment (shanemc-goog) [inline docs/contributor-guide/developer-guide/technology-overview.md:19]: Why delete this content completely?

File: docs/contributor-guide/developer-guide/testing.md [hunk 6]
```diff
@@ -225,17 +256,19 @@ GitHub action run.
 
 #### De-randomizing
 
-Timestamp/Dates and applicant IDs will change each time a test is ran, to
+Timestamp/Dates and applicant IDs will change each time a test is run, to
 automatically normalize these, UI elements that contain them need to have the
 [`ReferenceClasses`](https://sourcegraph.com/github.com/civiform/civiform/-/blob/server/app/views/style/ReferenceClasses.java?L133&subtree=true)
-`BT_DATE` and `BT_APPLICATION_ID` added respectively.
+`BT_DATE`, `BT_APPLICATION_ID` and few other classes added respectively.
```
Comment (shanemc-goog) [inline docs/contributor-guide/developer-guide/testing.md:262]:
```suggestion
`BT_DATE`, `BT_APPLICATION_ID` and a few others added respectively.
```

File: docs/contributor-guide/developer-guide/testing.md [hunk 7]
```diff
@@ -225,17 +256,19 @@ GitHub action run.
...
+Check [normalizeElements](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/civiform/civiform%24+normalizeElements&patternType=standard&sm=1)
```
Comment (shanemc-goog) [inline docs/contributor-guide/developer-guide/testing.md:263]:
```suggestion
Check the [normalizeElements](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/civiform/civiform%24+normalizeElements&patternType=standard&sm=1)
```

File: docs/contributor-guide/developer-guide/testing.md [hunk 8]
```diff
@@ -225,17 +256,19 @@ GitHub action run.
...
+function for up-to-date list of elements and update if necessary.
```
Comment (shanemc-goog) [inline docs/contributor-guide/developer-guide/testing.md:264]:
```suggestion
function for an up-to-date list of elements and update if necessary.
```

**Review by shanemc-goog — CHANGES_REQUESTED** [review]

File: docs/contributor-guide/developer-guide/technology-overview.md (hunk 5 above)
Comment (nbeloglazov) [reply docs/contributor-guide/developer-guide/technology-overview.md:19]: We don't use WebJars. Looks like we used for azure JS but later switched to loading JS from CDN. Going forward we are going to import JS libraries through webpack/npm and not WebJars.

File: docs/contributor-guide/developer-guide/testing.md (hunk 7 above)
Comment (nbeloglazov) [reply docs/contributor-guide/developer-guide/testing.md]: Done.

File: docs/contributor-guide/developer-guide/testing.md (hunk 6 above)
Comment (nbeloglazov) [reply docs/contributor-guide/developer-guide/testing.md]: Done.

**Review by shanemc-goog — APPROVED** [review]

**Pull request merged**